gpii/node_modules/preferencesServer/src/preferencesServerConst.js

```diff
@@ -24,7 +24,20 @@ gpii.preferencesServer.defaultPrefsSetId = "gpii-default";
 
 // Error messages
 gpii.preferencesServer.errors = fluid.freezeRecursive({
-    missingGpiiKey: "Missing GPII key",
-    unfoundGpiiKey: "Cannot find the GPII key \"%gpiiKey\"",
-    missingPreferences: "Missing preferences"
+    missingGpiiKey: {
+        errorCode: "GPII_ERR_NO_GPIIKEY",
+        message: "GPII key does not exist"
+    },
+    gpiiKeyExisted: {
+        errorCode: "GPII_ERR_GPIIKEY_ALREADY_EXIST",
+        message: "GPII key \"%gpiiKey\" already exists"
+    },
+    noUpdateOnSnapset: {
```

Comment:
Rather than encode "snapset" in this key and message name, I think it's wiser to just refer to the prefs set being "readOnly" (which it might be for a variety of reasons). GPII_ERR_PREFS_READONLY and similarly in the key name. You could refer to "snapset" in the message detail - e.g. "Can't update preferences for readOnly GPII key %gpiiKey (perhaps it is a snapset)"